tests/import_proxy_test.go

```diff
@@ -0,0 +1,326 @@
+)
+
+const (
+	username         = "foo"
+	password         = "bar"
+	httpPort         = "8080"
+	httpPortWithAuth = "8081"
+	proxyServerName  = "cdi-test-proxy"
+)
+
+var _ = Describe("Import Proxy tests", func() {
+
+	f := framework.NewFramework("import-proxy-func-test")
+	tinyCoreIsoHTTP := func() string { return fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs) }
+	tinyCoreIsoHTTPS := func() string { return fmt.Sprintf(utils.HTTPSTinyCoreIsoURL, f.CdiInstallNs) }
+	createHTTPDataVolume := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		return dataVolume
+	}
+	createHTTPDataVolumeWithAuth := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		stringData := map[string]string{
+			common.KeyAccess: username,
+			common.KeySecret: password,
+		}
+		s, _ := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil, stringData, nil, f.Namespace.Name, "mysecret"))
+		dataVolume.Spec.Source.HTTP.SecretRef = s.Name
+		return dataVolume
+	}
+	createHTTPSDataVolume := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		cm, err := utils.CopyFileHostCertConfigMap(f.K8sClient, f.Namespace.Name, f.CdiInstallNs)
+		Expect(err).To(BeNil())
+		dataVolume.Spec.Source.HTTP.CertConfigMap = cm
+		return dataVolume
+	}
+	createHTTPSDataVolumeWithAuth := func(dataVolumeName, size, url string) *cdiv1.DataVolume {
+		dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, size, url)
+		cm, err := utils.CopyFileHostCertConfigMap(f.K8sClient, f.Namespace.Name, f.CdiInstallNs)
+		Expect(err).To(BeNil())
+		dataVolume.Spec.Source.HTTP.CertConfigMap = cm
+		stringData := map[string]string{
+			common.KeyAccess: username,
+			common.KeySecret: password,
+		}
+		s, _ := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil, stringData, nil, f.Namespace.Name, "mysecret"))
+		dataVolume.Spec.Source.HTTP.SecretRef = s.Name
+		return dataVolume
+	}
+	createProxyURL := func(isHTTPS bool, withBasicAuth bool) string {
+		var auth string
+		port := httpPort
+		if withBasicAuth {
+			port = httpPortWithAuth
+			auth = fmt.Sprintf("%s:%s@", username, password)
+		}
+		return fmt.Sprintf("http://%s%s.%s:%s", auth, proxyServerName, f.CdiInstallNs, port)
+	}
+
+	FDescribe("importing from cdi proxied web server", func() {
+		type importProxyTestArguments struct {
+			name               string
+			size               string
+			url                func() string
+			proxyURL           func(bool, bool) string
+			isHTTPS            bool
+			proxyWithBasicAuth bool
+			noProxyDomains     string
+			dvFunc             func(string, string, string) *cdiv1.DataVolume
+			expected           func() types.GomegaMatcher
+		}
+
+		FDescribeTable("should", func(args importProxyTestArguments) {
```

Comment:
Just an FYI if you FDescribeTable, you don't have to FEntry each entry, you are already focusing on that particular table.